### PR TITLE
Add int8 and int16 support to tflu network tester.

### PR DESCRIPTION
**Pull request by sicong-li-arm**
Add support for other quantized types (int8, int16) to tflu network tester.

Signed-off-by: SiCong Li <sicong.li@arm.com>

### CODE REVIEW

**Comment by sicong-li-arm** [comment]
@petewarden Could you please review this patch? Thanks! :)

**Comment by sicong-li-arm** [comment]
@petewarden Could you please assign someone to review this patch? Thanks! :)

**Comment by sicong-li-arm** [comment]
@petewarden @njeffrie Could you please have a look into this patch? Thanks!

**Review by njeffrie — COMMENTED** [review]
A couple of high level questions and comments:
1. What model do we have to test with int8 / int16 quantization?  I would prefer to add features along with a use case.  Typically this would mean adding an int8 and int16 model to test, alongside network_model.cc. 
2. Why are you adding quantization scale / zero point to input / output tensors? This information should always be encoded in the model itself.
3. I'm a bit loathed to add another way to test each element in an array against an expected output. I see the need for a method to compare with templated types, but I feel that this should live in our common testing code, not in this test.  I recognize this is largely a product of my own messy attempts to write generic test helpers, so perhaps leaving it in here is OK until we clean up our testing story internally.

**Comment by gbaned** [comment]
@sicong-li-arm Can you please check @njeffrie's comments and keep us posted. Thanks!

File: tensorflow/lite/micro/examples/network_tester/expected_output_data.h [hunk 1]
```diff
@@ -16,6 +16,9 @@ limitations under the License.
 #ifndef TENSORFLOW_LITE_MICRO_EXAMPLES_NETWORK_TESTER_EXPECTED_OUTPUT_DATA_H_
 #define TENSORFLOW_LITE_MICRO_EXAMPLES_NETWORK_TESTER_EXPECTED_OUTPUT_DATA_H_
 
+static const bool output_use_default_quant_params = true;
+static const float output_scale = 0;     // NOT USED
```
[the remark this answers is not in the document]
Comment (sicong-li-arm) [reply tensorflow/lite/micro/examples/network_tester/expected_output_data.h]: Done removed.
It was a big misunderstanding from my part.

**Comment by sicong-li-arm** [comment]
> A couple of high level questions and comments:
> 
> 1. What model do we have to test with int8 / int16 quantization?  I would prefer to add features along with a use case.  Typically this would mean adding an int8 and int16 model to test, alongside network_model.cc.
> 2. Why are you adding quantization scale / zero point to input / output tensors? This information should always be encoded in the model itself.
> 3. I'm a bit loathed to add another way to test each element in an array against an expected output. I see the need for a method to compare with templated types, but I feel that this should live in our common testing code, not in this test.  I recognize this is largely a product of my own messy attempts to write generic test helpers, so perhaps leaving it in here is OK until we clean up our testing story internally.

@njeffrie Thank you for your review! Sorry again for the late reply. I'll keep a close watch to this PR this week.
1. This was partially related to [this PR](https://github.com/tensorflow/tensorflow/pull/38873) where we added int16 support to softmax. In general we'd like to extend the network tester test's ability to at least run a single int16 softmax layer.
2. Yep you're right. This was a misunderstanding on my side.

**Review by petewarden — DISMISSED** [review]
Looks good, thanks!

**Comment by rthadur** [comment]
@sicong-li-arm please resolve conflicts ,once done we can merge this PR ASAP.

**Comment by sicong-li-arm** [comment]
> @sicong-li-arm please resolve conflicts ,once done we can merge this PR ASAP.

@rthadur Done. Thanks! :)

**Review by rthadur — APPROVED** [review]
